torch/distributed/distributed_c10d.py

```diff
@@ -4394,6 +4394,7 @@ def _new_group_with_tag(
     global _world
 
     default_pg = _get_default_group()
+    device_id = default_pg.bound_device_id or None
```

Comment:
is it really necessary to have `or None`? `bound_device_id` property will already return `None` when unset

Reply:
Oops, yea this is kinda dumb. I will submit a cleanup PR.